### ISSUE TITLE
Meta: Bus factor

### ISSUE THREAD

**Issue by enterprisey** [libui-ng/libui-ng#77]
Could more people be added as committers to ensure project sustainability?

**Comment by rubyFeedback** [comment]
Agreed. I think this may be useful in the event that cody does not have time. For instance, Andy recently made a comment in regards to child/parent window, which is a libui issue ( https://github.com/kojix2/LibUI/issues/52 and https://github.com/andlabs/libui/issues/137 ). Perhaps we can add more code support if patches exist, and track / coordinate via libui-ng.

**Comment by szanni** [comment]
Thanks for opening this issue. I believe this should be _HIGH_ priority otherwise I do not see how libui-ng is any different to libui.

I am wondering what the best way forward would be. Many open source projects have a mailing list with patches +signed off type of ensuring patches are high enough quality. I am not aware of github offering such an option, which kind of surprises me.

Even I was thinking about forking the original project at some time, but I honestly don't have the time to do so, while I'm happy to review and contribute semi-regularly. As it stands it seems like @cody271  is in a similar boat. Last activity seems to be a month ago - which is perfectly fine. I would just wish for more communication and a possibility to move forward with others that have a similar ideal with regards to code quality and understanding of scope. Especially API changes/additions should undergo great scrutiny.

**Comment by cody271** [comment]
Don't worry everyone, libui-ng remains a high priority! At least until we are done with an initial 1.0 release. I've simply had to focus on another (paying) project for the past month. Next time I guess I'll make a formal announce as such. Also reminder that the libui-bindgen rewrite is still in progress but not yet public. Please have patience.

> Thanks for opening this issue. I believe this should be _HIGH_ priority otherwise I do not see how libui-ng is any different to libui.
> 

Absolutely. This was the entire point of creating the @libui-ng organization instead of using my personal account. The one firm guarantee that I can make for the long term here is that history won't repeat. If at some point in the future I can no longer work on the project whatsoever, I'll make that VERY clear and hand off any and all responsibilities immediately. But I have no such intentions as of this moment! To anyone who is serious about the responsibility of being added as a maintainer and is in a position to do so long term, should reach out to me directly. Thanks.

> I am wondering what the best way forward would be. Many open source projects have a mailing list with patches +signed off type of ensuring patches are high enough quality. I am not aware of github offering such an option, which kind of surprises me.
> 

IIRC GitLab does have some features like this, worth looking into further.

**Comment by szanni** [comment]
Thanks for replying cody. I know life gets in its way sometimes. I guess it was just a little concerning seeing no activity and/or responses. This project is a fairly big undertaking, I do not think anyone should expect 24/7 updates or responses. Just a matter of communication. I know myself how much of a time sink just submitting patches can be.

And the bus factor is not only people disappearing, but literally getting hit by a bus. In such an unfortunate event no handoff would even be possible, that’s why I believe we should still have more than one person with admin rights to the repo (in whatever way GitHub handles that). Just as a fallback even, somebody who has proven to have the projects best interest in mind.

> IIRC GitLab does have some features like this, worth looking into further.

That might be worth looking into. But if we have others with access to the repo that might be good enough for now. Maybe these could be things on a TODO list? And volunteers could pick up such tasks, research and make a proposal? Just an idea.

**Comment by norepimorphism** [comment]
Any updates on this?

As @szanni said, this is a big undertaking and no one (including me!) expects 24/7 updates or commit activity. That being said, it has been a couple months since the last commit. I would love to contribute, but I'm worried my issues/PRs might idle.

**Comment by cody271** [comment]
> Any updates on this?
> 
> As @szanni said, this is a big undertaking and no one (including me!) expects 24/7 updates or commit activity. That being said, it has been a couple months since the last commit. I would love to contribute, but I'm worried my issues/PRs might idle.
>

@norepimorphism No specific update to share, but PRs very much welcome!
If you don't receive a review of your PRs in a reasonable timeframe, please feel free to reach out and ping me directly. :)

(Especially anything Windows related will receive top priority, as we've fallen a bit behind on Windows somewhat in comparison to Unix/Darwin support I'm afraid..)

**Comment by norepimorphism** [comment]
Great! I was actually planning on looking at updating the Windows code, so that works out well. Thanks!
